shelley/chain-and-ledger/executable-spec/src/STS/Ledgers.hs

```diff
@@ -73,6 +73,7 @@ ledgersTransition = do
   let u''' = UTxOState utxo' dep fee (ppup, aup, favs', avs')
 
   let ds = _dstate dw''
+      ps = _pstate dw''
       fdms_    = _fdms ds
       Dms dms_ = _dms ds
       (fdms', curr) = Map.partitionWithKey (\(s, _) _ -> slot <= s) fdms_
```

Comment:
not part of the PR, but I think `curr` and `fdms'` is the wrong way, `fst` is the part that matches the predicate

Reply:
it's true that `fst` is the part that matches the predicate, but I think it's still correct.

`s` is the lot assigned to a future delegation, and `slot` is the current slot. so if `slot <= s`, then this "future" delegation is legitimately in the future. otherwise, it's now a current (ie should be considered now) delegation.